### PR TITLE
Fix recent coverity issues in imaging stuff and various other places

### PR DESCRIPTION
**Pull request by FedeMPouzols**
Fixes three issues in imaging stuff and a few other issues that had crept in in the last week or so before the code freeze. Leaving this for v3.8 to minimize risks.

Most of them are not to terrible (unitialized members that shouldn't be risky with the current code) but maybe the issue in `PropertyManager` could lead to confusing errors/crashes.

**To test:**
- Review changes
- All tests should pass

No issue number for this.

_Does not need to be in the release notes._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?
##### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/DataHandling/src/LoadNexusMonitors2.cpp [hunk 1]
```diff
@@ -68,7 +68,8 @@ void loadSampleDataISIScompatibilityInfo(
 } // namespace
 
 //------------------------------------------------------------------------------
-LoadNexusMonitors2::LoadNexusMonitors2() : Algorithm(), m_monitor_count(0) {}
+LoadNexusMonitors2::LoadNexusMonitors2()
+    : Algorithm(), m_monitor_count(0), m_allMonitorsHaveHistoData(false) {}
```
Comment (SimonHeybrock) [inline Framework/DataHandling/src/LoadNexusMonitors2.cpp]: Could use in-class member init here as well.

Comment (FedeMPouzols) [inline Framework/DataHandling/src/LoadNexusMonitors2.cpp]: That would be nicer, but I kept this for consistency. I was thinking that a more project-wide revamp of initializations and constructors could be a maintenance task, either for this or the next maintenance period.

**Comment by SimonHeybrock** [comment]
@FedeMPouzols Looks good. If you want to tidy up the constructors let me know, otherwise I will put the shipit.

**Comment by martyngigg** [comment]
@FedeMPouzols This has the milestone set to 3.8 but the target branch is `3.7`. I assume that it was meant to be `master`?

**Comment by FedeMPouzols** [comment]
Oops, yes. We changed plans to minimize risks. I can close/reopen against 3.8 if there's merge issues.

**Comment by martyngigg** [comment]
@FedeMPouzols Yes I think we should close this one and reopen another for `master`. As long as the builds pass we can merge and point at this for the main review.

**Pull request closed without merging**